From 73e3dbb60c698a3488149f79119f65d26c36a488 Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 13 Aug 2018 07:02:19 +0300
Subject: [PATCH] board: gaudi is a new ARM64 SoC Fixup

Update mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG to L/H
since Gaudi got 54 bits and not only 23 bits as dali

Also update comments to clear out that use is for Dali only.
---

diff --git a/board/habanalabs/common/btl/pcie_phy.s b/board/habanalabs/common/btl/pcie_phy.s
index f6f60f2..5e96b13 100644
--- a/board/habanalabs/common/btl/pcie_phy.s
+++ b/board/habanalabs/common/btl/pcie_phy.s
@@ -21,7 +21,7 @@
 cfg mmPCIE_PHY_CR_PARA_CFG, 0x00000001
 
 #ifdef mmPCIE_PHY_MEM_WR_MC
-/* Errata: disable unused design that cause PCIe phy SRAM DERR */
+/* Dali Errata: disable unused design that cause PCIe phy SRAM DERR */
 cfg mmPCIE_PHY_MEM_WR_MC, 0x0
 #endif
 
diff --git a/board/habanalabs/common/btl/primary.S b/board/habanalabs/common/btl/primary.S
index c5a36b2..f098c2a 100644
--- a/board/habanalabs/common/btl/primary.S
+++ b/board/habanalabs/common/btl/primary.S
@@ -11,7 +11,7 @@
 nop
 
 #ifdef mmPSOC_GLOBAL_CONF_WARM_REBOOT
-/* Set CPU_BOOT_STATUS_IN_BTL */
+/* Dali: Set CPU_BOOT_STATUS_IN_BTL */
 cfg mmPSOC_GLOBAL_CONF_WARM_REBOOT, 0x4
 #endif
 
@@ -21,6 +21,9 @@
 #ifdef mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG
 /* Configure PRSTN to reset entire SoC */
 cfg mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG, 0x7FFFFF
+#else
+cfg mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L, 0xFFFFFFFF
+cfg mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_H, 0x3FFFFF
 #endif
 
 /* Clear masking of PRSTN from SW reset */
